ws-client/src/helpers.rs

```diff
@@ -72,6 +72,33 @@ pub fn process_subscription_response(
 	}
 }
 
+/// Attempts to process an incoming notification
+///
+/// Returns Ok() if the response was successfully handled
+/// Returns Err() if there was no handler for the method
+
+pub fn process_notification(manager: &mut RequestManager, notif: JsonRpcNotifResponse<JsonValue>) -> Result<(), Error> {
+	let sub_id: SubscriptionId = SubscriptionId::Str(notif.method.to_owned());
+	let request_id = match manager.get_request_id_by_subscription_id(&sub_id) {
+		Some(request_id) => request_id,
+		None => return Err(Error::InvalidSubscriptionId),
+	};
+
+	match manager.as_notification_handler_mut(&request_id) {
+		Some(send_back_sink) => match send_back_sink.try_send(notif.params) {
+			Ok(()) => Ok(()),
+			Err(err) => {
+				log::error!("Error sending notification subscription {:?} error: {:?}", sub_id, err);
+				Err(Error::Internal(err.into_send_error()))
+			}
+		},
+		None => {
+			log::error!("Subscription ID: {:?} not an active subscription", sub_id);
+			Err(Error::InvalidSubscriptionId)
```

Comment:
This error is a bit confusing on the frontend so maybe we should have a specific error for unexpected notification

Reply:
Yeah, just add a new error `InvalidNotification`, `UnregisteredNotification` or similar.